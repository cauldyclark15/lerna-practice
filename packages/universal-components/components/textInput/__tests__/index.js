import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { mountToJson } from 'enzyme-to-json';
import InputComponent from '../';

Enzyme.configure({ adapter: new Adapter() });

describe('<InputComponent />', () => {
  it('<InputComponent />', () => {
    const wrapper = mount(<InputComponent onKeyPress={() => {}} />);
    expect(mountToJson(wrapper)).toMatchSnapshot();
  });

  it('onKeyPress()', () => {
    const spy = jest.fn();
    const wrapper = mount(<InputComponent onKeyPress={spy} />);
    wrapper
      .find('TextInput')
      .first()
      .props()
      .onKeyPress();

    expect(spy).toBeCalled();
  });
});
