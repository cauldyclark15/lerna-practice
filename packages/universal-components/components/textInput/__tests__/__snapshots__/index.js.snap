// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputComponent /> <InputComponent /> 1`] = `
<InputComponent
  onKeyPress={[Function]}
>
  <View>
    <div
      className="rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1pxmb3b rn-flexBasis-7vfszb rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
    >
      <TextInput
        autoCapitalize="sentences"
        autoComplete="on"
        autoCorrect={true}
        editable={true}
        keyboardType="default"
        multiline={false}
        numberOfLines={2}
        onKeyPress={[Function]}
        secureTextEntry={false}
        style={72}
      >
        <input
          autoCapitalize="sentences"
          autoComplete="on"
          autoCorrect="on"
          className="rn-appearance-30o5oe rn-backgroundColor-wib322 rn-borderBottomWidth-9cy4zw rn-borderTopColor-ln5t45 rn-borderRightColor-1kbry23 rn-borderBottomColor-bi4u99 rn-borderLeftColor-1eue9rr rn-borderTopLeftRadius-ou6ah9 rn-borderTopRightRadius-t12b5v rn-borderBottomRightRadius-zmljjp rn-borderBottomLeftRadius-pm2fo rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-16yz453 rn-font-1lw9tu2 rn-fontWeight-aruset rn-height-mabqd8 rn-letterSpacing-ud0q2t rn-paddingTop-m611by rn-paddingRight-1qfoi16 rn-paddingBottom-1mi0q7o rn-paddingLeft-1hfyk0a rn-resize-1dz5y72"
          dir="auto"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onSelect={[Function]}
          readOnly={false}
          type="text"
        />
      </TextInput>
    </div>
  </View>
</InputComponent>
`;
